### ISSUE TITLE
Build failure: visidata

### ISSUE THREAD

**Issue by victorhooi** [NixOS/nixpkgs#326083]
### Steps To Reproduce

1. Using [this](https://github.com/dustinlyons/nixos-config) as my Nix starter template for macOS - uses home-manager and nix-darwin.
2. Run `nix flake update`
3. Run `nix run ./#build-switch`, which causes some kind of build failure with visidata.

(Previously visitdata seemed to be working correctly - however, I recently ran a `nix flake update` - and it seems to have been broken for the last week or so).

### Build log

```
❯ nix run ./#build-switch
warning: Git tree '/Users/foobar/nixos-config' is dirty
Running build-switch for aarch64-darwin
Starting build...
warning: Git tree '/Users/foobar/nixos-config' is dirty
error: builder for '/nix/store/x63ii5nwcjsymn6lja476456ilnz06sz-visidata-3.0.2.drv' failed with exit code 1;
       last 10 log lines:
       > index 3a5decf..5c68fcc 100644
       > --- a/tests/golden/edit-type.tsv
       > +++ b/tests/golden/edit-type.tsv
       > @@ -4,5 +4,5 @@ Customer     0       0       17
       >  SKU        0       0       34                                              
       >  Item 0       0       40                                              
       >  Quantity     0       0       10      1       144     403     2       7.90    21.57
       > -Unit    0       0       28      0.0     2495.99 11974.530000000006      12.95   234.79  585.85
       > +Unit   0       0       28      0.0     2495.99 11974.53        12.95   234.79  585.85
       >  Paid   0       0       36      -2300.0 3602.1  10886.21        51.8    213.46  819.74
       For full logs, run 'nix log /nix/store/x63ii5nwcjsymn6lja476456ilnz06sz-visidata-3.0.2.drv'.
error: 1 dependencies of derivation '/nix/store/xvkdbn9frapnpkdqd5r5dvgvmawg4kid-home-manager-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/f876nb0z7z11lyd59aa95c8blk3a9ci9-home-manager-fonts.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8d79rdgzvd3jdkw70g04aqdjhhhrdwcj-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5i47v39n5kyz4y0sxcqbk96dhnaf8g6z-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/82w9pvr6h87y5rlzmafwyqqpx54xkz6v-system-applications.drv' failed to build
error (ignored): error: cannot unlink '/private/tmp/nix-build-swift-5.8.drv-4/src/llvm-project/clang/test/CodeGenCXX': Directory not empty
error: 1 dependencies of derivation '/nix/store/hyb9d89pppijms5c773ikivlwrnp66qj-visidata-3.0.2-fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6kfi2yd4dnji5kvj3h19scrx425mnpza-darwin-system-24.11.20240708.655a58a+darwin4.cf297a8.drv' failed to build
```

I've uploaded the build logs (`nix log /nix/store/x63ii5nwcjsymn6lja476456ilnz06sz-visidata-3.0.2.drv > /tmp/visidata_nix_build_logs`) as a gist here:

https://gist.github.com/victorhooi/95531bfe36848d45ed54cadb41269fb1

### Notify maintainers

@markus1189 @pacien

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

I get this error when I try to run this command:

```console
❯ nix-shell -p nix-info --run "nix-info -m"
error:
       … while calling the 'import' builtin
         at «string»:1:18:
            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (nix-info) ]; } ""
             |                  ^

       … while realising the context of a path

       … while calling the 'findFile' builtin
         at «string»:1:25:
            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (nix-info) ]; } ""
             |                         ^

       error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)

nixos-config on  main [!] 
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by markus1189** [comment]
Looking at the build log, that is a failing test from visidata itself, where something is wrong with the number having a different value `11974.530000000006` vs `11974.53`

So that should hopefully be fixed upstream

**Comment by victorhooi** [comment]
Well, that's what it looked like at first to my naive eyes (a failing test).

However, it didn't make sense because it's Visidata 3.0.2 - which was released back in January 2024. Also, if I look at the Nix visidata package - there's been no changes in 6 months.

And I could have sworn visidata was working for me last week or so, before I ran `nix flake update` since then...although maybe I'm mistaken...

**Comment by anjakefala** [comment]
Hi @victorhooi!

Do you, by any chance have `options.disp_float_fmt` set to anything?

**Comment by anjakefala** [comment]
That gives me the info to address this!

**Comment by galenhuntington** [comment]
This is fixed upstream.

**Comment by victorhooi** [comment]
Thanks @anjakefala and @pacien! =).

Quick question for a Nix newbie - if I'm currently using nixpkgs-unstable (default on the [templates](https://github.com/dustinlyons/nixos-config) I'm using) - but Visidata hasn't cut a new release yet with this fix just yet - and nixpkgs is using 3.0.2 ([visidata default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/visidata/default.nix)) - is there an easy way to have it build from the visidata develop branch instead?

Or is it just a case of waiting for the next visidata release, and then also for the nixpkgs to get updated?
